### DISCUSSION TITLE
using volt.img tag in INVO.

### DISCUSSION THREAD

**Discussion by emagus** [phalcon/cphalcon#16247] [Q&A] [answered]
Hi, 
Installing INVO in http://localhost/invo/. (xampp. Apache, mod_rewrite, php 8.1, Phalcon 5.1.2) using volt for tag img :

{{ img('assets/agency.png' }}

When the route is http://localhost/invo/ is ok.(http://localhost/invo/assets/agency.png)
when the route is another like http://localhost/invo/about/index, the route for img is (http://localhost/invo/about/assets/agency.png) and is wrong.

index.php : (i've changed becouse not working ) $_SERVER['REQUEST_URI']
```
    (new Application($di))
        ->handle($_GET['_url'] ?? '/')
        ->send()
```

```
    'application' => [
        'viewsDir' => $_ENV['VIEWS_DIR'] ?? 'themes/invo',
        'baseUri'  => $_ENV['BASE_URI'] ?? '/invo/',
    ],
```

I think is similar to question "brand new application, controller not found?"

**Comment by s-ohnishi** [comment, marked as answer]
I believe the sample application invo is configured to put a set of downloaded files in the document root.
So, if you extract it to a directory placed in the document root like you did, various inconsistencies will occur.
for example,
this line
```
         ->handle($_SERVER['REQUEST_URI'])
```
changed to
```
         ->handle($_GET['_url'] ?? '/')
```
is required, otherwise the top page cannot be opened.

Other reasons for thinking that it is assumed to be placed directly in the document root are:
in the template,
The link to this top page is
```
<a class="navbar-brand" href="/">INVO</a>
```
I think you have to write it via url().
(For example,
```
{{ tag.a(url('/'), 'INVO', ['class':'navbar-brand']) }}
```
)


```
{% set topMenu = [
     'index': [
         'title': 'Home',
         'uri': '/index',
         'with_auth': false
     ],
     'invoices': [
         'title': 'Invoices',
         'uri': '/invoices/index',
         'with_auth': true
     ],
     'about': [
         'title': 'About',
         'uri': '/about/index',
         'with_auth': false
     ],
     'contact': [
         'title': 'Contact',
         'uri': '/contact/index',
         'with_auth': false
     ]
] %}
```
It is defined like this and used as follows.
```
             {% for controller, menu in topMenu %}
                 {% if (session.has('auth') and menu['with_auth'] === true) or menu['with_auth'] === false %}
                 <li class="nav-item {% if controller == dispatcher.getControllerName()|lower %}active{% endif %}">
                     <a class="nav-link" href="{{ menu['uri'] }}">{{ menu['title'] }}</a>
                 </li>
                 {% endif %}
             {% end for %}
```

It is recommended to stop using it in subdirectories and put it in the document root.
Alternatively, I think it's good to access with a different host name assigned a document root different from the default with a virtual host.
Otherwise, I think it will be necessary to make major modifications before using it as a sample.
And unfortunately, the current invo-2.0.0 is not compliant with Phalcon v5, so it will need to be modified accordingly.

**Reply by emagus** [reply]
solved using {{ tag.img(url('assets/media/auth/agency.png') }} 

Thanks @s-ohnishi
